uml/convert_current_state.puml

```diff
@@ -0,0 +1,50 @@
+@startuml
+title Conversion
+
+actor User
+
+box MAT
+participant MAT
+participant "MAT DB"
+end box
+
+box FHIR
+participant MicroService
+participant "MicroService's MongoDB"
+participant "HAPI FHIR"
+participant VSAC
+end box
+
+User -> MAT: find and select a measure
+MAT -> "MAT DB": pull measure records
+"MAT DB" --> MAT: records
+MAT -> MAT: if measure is convertible, \nenable convert button
+MAT --> User: measure selected
+User -> MAT: convert selected measure
+MAT -> MicroService: validate\n(measureID, type=VALIDATION, format=SIMPLE)
+MicroService -> MicroService: validate
+MicroService -> VSAC: value set validation
+MicroService -> "MicroService's MongoDB": store report
+MicroService --> MAT: report
+MAT --> User: [not valid] return, redirect to show report
+MAT -> MAT: [valid] proceed with conversion
+MAT -> MAT: clone source measure \ninto new draft measure \nwith FHIR type
+MAT -> "MAT DB": persist draft FHIR measure
+"MAT DB" --> MAT: fhirMeasureId
+MAT -> MicroService: convert\n(measureId, type=CONVERSION, format=SIMPLE)
+MicroService -> MicroService: validate
+MicroService -> VSAC: value set validation
+MicroService -> "MicroService's MongoDB": store report
```

Comment:
We are already Validating and storing report in MongoDB right? After Cloning & validating, will it be stored as a new report ?

Reply:
It was not the intention to provide a technical decision or a future design. I just documented some of the flows in PlantUML format.
You can render sequence diagrams out of that with a PlantUML plugin (Intellij or Confluence).

Reply:
as far as I know, the report is already stored in MongoDB on the micro-service end.

Reply:
Yup I did get PlantUML